### PR TITLE
Add docs on indexing dataobjects

### PR DESCRIPTION
**Pull request by chillu**
Copied from CWP docs at https://www.cwp.govt.nz/guides/core-technical-documentation/common-web-platform-core/en/cwp-features/solr_search.
It's not specific to this platform, so should be part of the general docs.

### CODE REVIEW

File: docs/en/Solr.md [hunk 1]
```diff
@@ -345,6 +345,71 @@ Now apply the configuration:
...
+		
+	}
+
+This is a copy/paste of the existing configuration but with the addition of `StaffMember`.
+
+Once you've created the above classes and run `flush=1`, access `dev/tasks/Solr_configure` and `dev/tasks/Solr_reindex`
+to tell Solr about the new index you've just created. This will add `StaffMember` and the text fields it has to the
+index.
+
+Now in your `mysite/_config.php` file, add the following:
```
Comment (tractorcow) [inline docs/en/Solr.md]: This code is specific to CWP and shouldn't be here. :D

Comment (chillu) [inline docs/en/Solr.md]: @tractorcow Which code is CWP specific in here?

Comment (tractorcow) [inline docs/en/Solr.md]: All of the code you pasted below. :D

**Comment by chillu** [comment]
@tractorcow I've removed the CWP specific docs, thanks for catching that!

File: docs/en/Solr.md [hunk 2]
```diff
@@ -378,6 +378,62 @@ Now apply the configuration:
...
+`getShowInSearch` is required to get the record to show in search, since all results are filtered by `ShowInSearch`.
+
+So with that, let's create a new class called `MySolrSearchIndex`:
+
+	:::php
+	<?php
+	class MySolrSearchIndex extends SolrIndex {
+		
+		public function init() {
+			$this->addClass('BasePage');
```
Comment (tractorcow) [inline docs/en/Solr.md]: I'd replace this with `SiteTree`. When adding dataobjects, it's best to add the base dataobject class if possible.

File: docs/en/Solr.md [hunk 3]
```diff
@@ -378,6 +378,62 @@ Now apply the configuration:
...
+			
+			$this->addAllFulltextFields();
+			$this->addFilterField('ShowInSearch');
+		}
+		
+	}
+
+This is a copy/paste of the existing configuration but with the addition of `StaffMember`.
+
+Once you've created the above classes and run `flush=1`, access `dev/tasks/Solr_configure` and `dev/tasks/Solr_reindex`
```
Comment (tractorcow) [inline docs/en/Solr.md]: Might be safest to use the correct case on these actions; They are `Solr_Configure` and `Solr_Reindex`.

**Comment by tractorcow** [comment]
Sorry to always add problems. :D I hope you don't mind me spotting a few other things.

**Comment by tractorcow** [comment]
Ping @chillu :D

**Comment by dhensby** [comment]
@chillu - do you have any desire to revisit this and push it over the line?

**Comment by dhensby** [comment]
Meh, merged by hand and made the ammends

**Pull request merged**